cli/src/diff_util.rs

```diff
@@ -703,25 +707,36 @@ pub fn show_file_by_file_diff(
                     writeln!(formatter, " {source}")?;
                     continue;
                 }
-                _ => {}
+                _ => {
+                    items.push((ui_path, path, (left_value, right_value)));
+                }
             }
-            let left_path = create_file(&path, &left_wc_dir, left_value)?;
-            let right_path = create_file(&path, &right_wc_dir, right_value)?;
-
-            invoke_external_diff(
-                ui,
-                formatter.raw(),
-                tool,
-                maplit::hashmap! {
-                    "left" => left_path.to_str().expect("temp_dir should be valid utf-8"),
-                    "right" => right_path.to_str().expect("temp_dir should be valid utf-8"),
-                },
-            )
-            .map_err(DiffRenderError::DiffGenerate)?;
         }
         Ok::<(), DiffRenderError>(())
     }
-    .block_on()
+    .block_on()?;
+    let num_items = items.len();
+    for (idx, (ui_path, path, (left_value, right_value))) in items.into_iter().enumerate() {
+        writeln!(
+            formatter.labeled("header"),
+            "JJ (#{} of {}): comparing {ui_path}",
```

Comment:
Could you put another newline before `JJ`?

--------------

My reasoning for the header is to make it clearer what's going on

- if there's an error
- if a GUI tool is called each time

It's not perfect, and perhaps other people will have other opinions.

Reply:
Done

Reply:
Oh, I made a silly error. We shouldn't print that newline on the first iteration.

Reply:
Done